packages/apps/web-viewer-test/src/providers/EcsqlWidget.tsx

```diff
@@ -0,0 +1,147 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useActiveIModelConnection } from "@itwin/appui-react";
+import { QueryRowFormat } from "@itwin/core-common";
+import type { IModelConnection } from "@itwin/core-frontend";
+import {
+  Button,
+  ExpandableBlock,
+  LabeledTextarea,
+  Table,
+} from "@itwin/itwinui-react";
+import React, { useState } from "react";
+
+export default function EcsqlWidget() {
+  const [input, setInput] = useState("");
+  const [data, setData] = useState<Record<string, string | React.ReactNode>[]>(
+    []
+  );
+  const [headers, setHeaders] = useState<string[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState("");
+  const iModelConnection = useActiveIModelConnection() as IModelConnection;
```

Comment:
shouldnt need to do a cast here, its of type `IModelConnection | undefined` which is ok

Reply:
If I don't do a cast, I might need the `!` operator, right?
![image](https://user-images.githubusercontent.com/51834103/185499292-a8f115f3-8472-485a-af01-46dad5d29761.png)

Reply:
look at my next comment, do a null check earlier

Reply:
what happens if it was actually undefined and you cast it to an iModelConnection?

Reply:
You're right, that would be bad.